### DISCUSSION TITLE
Protocol: add "Content-type" header to "queries" POST request

### DISCUSSION THREAD

**Discussion by brcarp** [algolia/instantsearch#6234] [Ideas]
This isn't quite a bug or a feature request, but if the POST requests to the `queries` endpoint could include a `Content-Type: application/json` header, it might help browsers and debuggers parse the payload from the user to aid in debugging applications using `instantsearch`. Some tools trip up on this and expect, absent any cues from `Content-type`, that the payload is `application/x-www-form-urlencoded` (because some of the properties like `params` _are_ in fact URL-encoded) and presents it improperly parsed to the investigator.

**Comment by Haroenv** [comment]
This is done on purpose, to ensure the request is a "Simple Request" which doesn't need the CORS preflight request, you can learn more about that here: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests

It's also annoying for us to debug (we're thinking about transitioning to a plain text request in v5), but in the mean time I also made a tool to inspect the parameters: https://haroen.me/algolia-tools/#request-explorer
